streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -452,9 +463,14 @@ private void runLoop() {
                         consumer.seekToEnd(ex.partitions());
                         log.info(String.format("stream-thread [%s] setting topic to consume from latest offset %s", this.getName(), partition.topic()));
                     } else {
-                        //If not defined go to default config of earliest
-                        consumer.seekToBeginning(ex.partitions());
-                        log.info(String.format("stream-thread [%s] no setting defined for topic %s setting to consume from earliest offset", this.getName(), partition.topic()));
+                        if (originalReset != null) { //client could specify "none" and not provide custom offset resets
+                            if (originalReset.equals("earliest")) {
+                                consumer.seekToBeginning(ex.partitions());
+                            } else if (originalReset.equals("latest")) {
+                                consumer.seekToEnd(ex.partitions());
+                            }
+                            log.info(String.format("stream-thread [%s] no custom setting defined for topic %s using original config  %s for offset reset", this.getName(), partition.topic(), originalReset));
+                        }
```

Comment:
We should add `else` and throw `StreamsException` with cause using original `NoOffsetForPartitionException` (ie, `ex`). Furthermore, the error message should explain in detail what happened and how a user can fit it. Something like:
```
No valid committed offset found for input topic T (partition P) and no valid reset policy configured. You need to set configuration parameter "auto.offset.reset" or specify a topic specific reset policy via KStreamBuilder#stream(...) or KStreamBuilder#table(...).
```

The reason for throwing an exception is, that Streams has no change to start reading a topic/partition in a meaningful way. The user needs to fix this issue! This is also current behavior, because `NoOffsetForPartitionException` would be raise for `auto.offset.reset=none` and missing committed offsets, too.